group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java

```diff
@@ -367,6 +383,26 @@ public GroupCoordinatorConfig(AbstractConfig config) {
                 SHARE_GROUP_HEARTBEAT_INTERVAL_MS_CONFIG, SHARE_GROUP_SESSION_TIMEOUT_MS_CONFIG));
     }
 
+    @SuppressWarnings("unchecked")
+    private static List<ConsumerGroupPartitionAssignor> consumerGroupAssignors(
+        AbstractConfig config
+    ) {
+        // In unit tests, it is pretty convenient to have the ability to pass instantiated
```

Comment:
Yes, using the same object is indeed convenient for testing. However, it still acts as a backdoor. Perhaps we could shift this behavior out of production and into the `GroupMetadataManagerTestContext`, allowing it to override the `GroupCoordinatorConfig` to return the object directly. What do you think?

Reply:
Let me see what I can do.

Reply:
I just pushed a change. Let me know what you think.